lib/msgfmt/msgfmt.go

```diff
@@ -140,6 +140,22 @@ func findUserInputEndIdx(userInputStartIdx int, msg []rune, userInput []rune) in
 	return msgIdx
 }
 
+// skipTrailingInputBoxLine skips the next line if it contains any of the markers.
+// In case of Gemini and Cursor, the user input is echoed back in a box
+// This function searches for the markers passed by the caller and skips the next line if it contains all of them.
+func skipTrailingInputBoxLine(lines []string, lastUserInputLineIdx *int, markers ...string) {
+	if *lastUserInputLineIdx+1 >= len(lines) {
+		return
+	}
+	line := lines[*lastUserInputLineIdx+1]
+	for _, m := range markers {
+		if !strings.Contains(line, m) {
+			return
+		}
+	}
+	*lastUserInputLineIdx++
```

Comment:
I'm not a fan of spookily mutating `lastUserInputLineIdx` at a distance with the pointer. I suggest we instead modify the function signature of `skipTrailingInputBoxLine` to return `idx int, found bool`.